probeinterface/probe.py

```diff
@@ -0,0 +1,259 @@
+        
+        # electrode position and shape : handle with arrays
+        self.electrode_positions = None
+        self.electrode_plane_axes = None
+        self.electrode_shapes = None
+        self.electrode_shape_params = None
+        
+        # vertices for the shape of the probe
+        self.probe_shape_vertices = None
+        
+        # this handle the wiring to device : channel index on device side.
+        # this is due to complex routing
+        self.device_channel_indices = None
+        
+        # the Probe can belong to a ProbeBunch
+        self._probe_bunch = None
+    
+    def get_electrode_count(self):
+        """
+        Return how many electrodes on the probe.
+        """
+        assert self.electrode_positions is not None
+        return len(self.electrode_positions)
+    
+    def set_electrodes(self, positions=None, plane_axes=None, shapes='circle',
+                shape_params={'radius': 10}):
+        """
+        Parameters
+        ----------
+        positions :array (num_electrodes, ndim)
+            Posisitions of electrodes.
+        
+        plane_axes:  (num_electrodes, 2, ndim)
+            This defines the electrod plane (2d or 3d)
+            
+        shapes: scalar or array in 'circle'/'square'/'rect'
+            Shape for each electrodes.
+        
+        shape_params dict or list of dict
+            Contain kargs for shapes ("radius" for circle, "width" for sqaure, "width/height" for rect)
+        """
+        assert positions is not None
+        
+        positions = np.array(positions)
+        if positions.shape[1] != self.ndim:
+            raise ValueErrorr('posistions.shape[1] and ndim do not match!')
+        
+        self.electrode_positions = positions
+        n = positions.shape[0]
+        
+        # This defines the electrod plane (2d or 3d) where the electrode lies.
+        # For 2D we make auto
+        if plane_axes is None:
+            if self.ndim ==3:
+                raise ValueError('you need to give plane_axes')
+            else:
+                plane_axes = np.zeros((n, 2, self.ndim))
+                plane_axes[:, 0, 0] = 1
+                plane_axes[:, 1, 1] = 1
+        plane_axes = np.array(plane_axes)
+        self.electrode_plane_axes = plane_axes
+
+        # shape
+        if isinstance(shapes, str):
+            shapes = [shapes] * n
+        shapes = np.array(shapes)
+        if not np.all(np.in1d(shapes, _possible_electrode_shapes)):
+            raise ValueError(f'Electrodes shape must be in {_possible_electrode_shapes}')
+        if shapes.shape[0] !=n:
+            raise ValueError(f'Electrodes shape must have same length as posistions')
+        self.electrode_shapes = np.array(shapes)
+        
+        # shape params
+        if isinstance(shape_params, dict):
+            shape_params = [shape_params] * n
+        self.electrode_shape_params = np.array(shape_params)
+    
+    def set_shape_vertices(self, shape_vertices):
```

Comment:
I don't really understand what shape vertices are. Are these vertices just boundary points?

Reply:
Yes this is vertice for plotting the poligon of the probe shape.